### ISSUE TITLE
Handle dropped and replaced transactions v 2.0

### ISSUE THREAD

**Issue by sparrowDom** [OriginProtocol/origin-dollar#483]
For the transaction side bar view in the `/mint` part of the dapp we already implemented the logic so that the dapp is able to handle dropped&replaced transactions --> [see issue & PR](https://github.com/OriginProtocol/origin-dollar/issues/259)

It would be cool if we could develop a utility function that can listen for transaction being mined, and also be able to handle that transaction being dropped and replaced / speed up with more gwei. 

We could then use that on parts of the dapp not yet supporting it:
- approve stable coin before minting
- staking OGN
- claiming staked OGN
- claiming OGN compensation
- claiming OUSD compensation

**Comment by micahalcorn** [comment]
I was confused by this today when swapping with one of the AMM contracts. I suspect that [the original update](https://github.com/OriginProtocol/origin-dollar/issues/259) to handle dropped and replaced transactions only works with the Origin Vault transactions? If so, we should add the utility function, as you mentioned, to work with any transactions that go through the DApp.

**Comment by sparrowDom** [comment]
Yup that is right, it only subscribes to Origin Vault events right now
